src/NServiceBus.AcceptanceTests/Sagas/When_saga_handles_messages_concurrently.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_saga_handles_messages_concurrently : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public Task Should_complete_saga_eventually()
+        {
+            return Scenario.Define<Context>(c => { c.SomeId = Guid.NewGuid().ToString(); })
+                .WithEndpoint<ConcurrentHandlerEndpoint>(b =>
+                {
+                    b.When((session, context) =>
+                    {
+                        var t1 = session.SendLocal(new StartMessageOne
+                        {
+                            SomeId = context.SomeId
+                        });
+                        var t2 = session.SendLocal(new StartMessageTwo
+                        {
+                            SomeId = context.SomeId
+                        });
+                        return Task.WhenAll(t1, t2);
```

Comment:
to make this test fail (before the fix), it would require a race condition of the saga being started concurrently. Task.WhenAll doesn't guarantee that we hit that race condition, but using some wait handles would. Let's have a call if you can't make sense out of my words :D